### PR TITLE
Matomo application logs can now be written in syslog and errorlog

### PR DESCRIPTION
**Pull request by mwithheld**
### Description:

Issue #9400 Add log handlers syslog and errorlog; Add config for syslog ident; Document options.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by mwithheld** [comment]
This is an update to PR #17745

File: plugins/Monolog/config/config.php [hunk 1]
```diff
@@ -97,6 +99,13 @@
     'Piwik\Plugins\Monolog\Handler\FileHandler' => DI\create()
         ->constructor(DI\get('log.file.filename'), DI\get('log.level.file'))
         ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+    
+    '\Monolog\Handler\ErrorLogHandler' => DI\create()
+        ->constructor(null, DI\get('log.level.errorlog'))
```
Comment (diosmosis) [inline plugins/Monolog/config/config.php]: Looking at the ErrorHandler constructor, it seems like this shouldn't be set to null. If you want to only set the level parameter, and use defaults for the rest, you can use:

```
->constructorParameter('level', DI\get('log.level.errorlog'))
```

**Comment by diosmosis** [comment]
@mwithheld Thanks for fixing the issues! I left one other comment. After it's addressed it should be good to merge.

File: plugins/Monolog/config/config.php (hunk 1 above)
Comment (mwithheld) [reply plugins/Monolog/config/config.php]: That is one nifty dependency injection system you've got going!  So would it be like this?
```
   '\Monolog\Handler\ErrorLogHandler' => DI\create()
   ->constructorParameter('level', DI\get('log.level.errorlog'))
   ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
```

Thanks so much for the guidance dizzy!

Comment (diosmosis) [reply plugins/Monolog/config/config.php]: @mwithheld yes, that looks correct! And yes it is definitely a nifty DI system! The library is called php-di and the creator, @/mnapoli, worked w/ us for a while so our integration works rather well.

**Comment by mwithheld** [comment]
Added commit 14992a8 ErrorLogHandler constructor: Avoid null

File: plugins/Monolog/config/config.php [hunk 2]
```diff
@@ -97,6 +99,13 @@
     'Piwik\Plugins\Monolog\Handler\FileHandler' => DI\create()
         ->constructor(DI\get('log.file.filename'), DI\get('log.level.file'))
         ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+    
+    '\Monolog\Handler\ErrorLogHandler' => DI\create()
+        ->constructorParameter('level', DI\get('log.level.errorlog'))
+        ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+
+    '\Monolog\Handler\SyslogHandler' => DI\create()
+        ->constructor(DI\get('log.syslog.ident'), 'syslog', DI\get('log.level.syslog')),
```
Comment (diosmosis) [inline plugins/Monolog/config/config.php]: Apologies for the back and forth, but one last question here: was it intentional not to use `->method('setFormatter', DI\get('log.lineMessageFormatter.file'))` here?

**Comment by mwithheld** [comment]
Good catch - the discussion is welcome :)  That was not intended - I'm just
not used to using GitHub for editing PRs.

**Comment by diosmosis** [comment]
@mwithheld I've merged the PR, thanks for going through the trouble of creating the PR and following up on it! The contribution is greatly appreciated.

**Comment by mattab** [comment]
@diosmosis could you please document this new capability in https://matomo.org/faq/troubleshooting/faq_115/ ? Thanks

**Comment by diosmosis** [comment]
documented

File: plugins/Monolog/config/config.php [hunk 3]
```diff
@@ -97,6 +99,14 @@
     'Piwik\Plugins\Monolog\Handler\FileHandler' => DI\create()
         ->constructor(DI\get('log.file.filename'), DI\get('log.level.file'))
         ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+    
+    '\Monolog\Handler\ErrorLogHandler' => DI\autowire()
+        ->constructorParameter('level', DI\get('log.level.errorlog'))
+        ->method('setFormatter', DI\get('log.lineMessageFormatter.file')),
+
+    '\Monolog\Handler\SyslogHandler' => DI\create()
+        ->constructor(DI\get('log.syslog.ident'), 'syslog', DI\get('log.level.syslog'))
```
Comment (orlitzky) [inline plugins/Monolog/config/config.php:108]: Is this logging to the "syslog" facility? I would guess that "user" is most appropriate, but "syslog" is definitely not correct =)

(Discovered while trying to match on the facility and program name using syslog-ng.)

Lazy reference: https://en.wikipedia.org/wiki/Syslog#Facility

Comment (mwithheld) [reply plugins/Monolog/config/config.php:108]: You are correct this should be 'user'.
Should probably use 
```
->constructorParameter('ident'=>DI\get('log.syslog.ident'))
->constructorParameter('level'=>DI\get('log.level.syslog'))
```

**Comment by diosmosis** [comment]
Hi @mwithheld, thanks for creating the change so quickly! Would you be able to put it in a new pull request? This one has already been merged so it's no longer useful to push here.

**Comment by orlitzky** [comment]
Ok, thanks! And thanks for adding syslog support in the first place. Obviously, I was waiting for it =)

**Pull request merged**